### ISSUE TITLE
Macros goes on and on

### ISSUE THREAD

**Issue by naumanafsar** [VSCodeVim/Vim#3365]
**Describe the bug**
When I simply apply a Macro on more than the number of lines in the file, the macros don't stop at the end of the file but goes on and on. eg. If the number of lines is 5 and I apply it 10 times i.e `10@@`, It won't stop on the 5th line, but repeat itself 10 times on the last line of the file.

**To Reproduce**

1.  Record a macro like `qq <Shift>I<p><Esc><Shift>A</p>jq`
2.  Apply it on the first line `@q`
3.  Then again all the lines of the file `10@@`
4.  See the last line

**Expected behavior**
It is supposed to be stopped on the 5th line.

**Screenshots**

![vscode](https://user-images.githubusercontent.com/25094885/51085594-93c3f080-175d-11e9-9385-e222bf70f252.png)

- Extension (VsCodeVim) version: 1.0.1
- VSCode version: 1.30.2
- OS: Linux Mint 19 Tara x86_64

**Comment by J-Fields** [comment]
How does vim know when to stop, exactly? Is it when a motion tries to take you to a location that doesn't exist? (Like `j` on the last line?)

**Comment by roberestarkk** [comment]
Yes, if Vim tries to 'go' somewhere it can't, it'll fail a macro.
I've noticed that VSCode also doesn't 'fail' the macro when it contains a search that returns no results, where Vim also does this.

**Comment by ChristopherSchoening** [comment]
As far as I can see this is already implemented (there's also a test using the `f` motion which works correctly) but a motion (other than with `f`) to an invalid place doesn't update the state (`lastMovementFailed`) accordingly which results in the macro loop not breaking.
Maybe you can assume that when the cursor position after the motion remains the same that the motion failed because the intention of a motion is to move the cursor.
